sdk/src/main/scala/com/horizen/account/mempool/MempoolMap.scala

```diff
@@ -0,0 +1,149 @@
+package com.horizen.account.mempool
+
+import com.horizen.SidechainTypes
+import com.horizen.account.transaction.EthereumTransaction
+import scorex.util.{ModifierId, ScorexLogging}
+
+import java.math.BigInteger
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable
+import scala.util.Try
+
+case class MempoolMap() extends ScorexLogging {
+
+  val all = TrieMap[ModifierId, SidechainTypes#SCAT]()
+  val executableTxs = TrieMap[SidechainTypes#SCP, scala.collection.mutable.SortedMap[BigInteger, ModifierId]]()
+  val nonExecutableTxs = TrieMap[SidechainTypes#SCP, scala.collection.mutable.SortedMap[BigInteger, ModifierId]]()
+  val nonces = TrieMap[SidechainTypes#SCP, BigInteger]()
```

Comment:
Let's sync about responsibilities of AccountMempool and MempoolMap.
As for me `MempoolMap` expose must encapsulate more.
At the moment the AccountMempool has a direct access to `all`, `executables` and others.
Check the actual implementation for UTXO model: https://github.com/HorizenOfficial/Sidechains-SDK/blob/dev/sdk/src/main/scala/com/horizen/SidechainMemoryPool.scala

mempool knows nothing about the internals of the MempoolMap

Reply:
Done